src/meta/src/hummock/compaction/manual_compaction_picker.rs

```diff
@@ -61,30 +62,41 @@ impl CompactionPicker for ManualCompactionPicker {
         tmp_sst_info.key_range = Some(self.option.key_range.clone());
         range_overlap_info.update(&tmp_sst_info);
 
-        let level_table_infos: Vec<SstableInfo> = levels[level]
-            .table_infos
-            .iter()
-            .filter(|sst_info| range_overlap_info.check_overlap(sst_info))
-            .filter(|sst_info| {
-                if self.option.internal_table_id.is_empty() {
-                    return true;
-                }
-
-                // to collect internal_table_id from sst_info
-                let table_id_in_sst: Vec<u32> =
-                    sst_info.get_table_ids().iter().cloned().collect_vec();
-
-                // to filter sst_file by table_id
-                for table_id in &table_id_in_sst {
-                    if self.option.internal_table_id.contains(table_id) {
+        let level_table_infos: Vec<SstableInfo> = if !self.option.sst_ids.is_empty() {
```

Comment:
can we use the filter chain instead of if else ?

just like key_range, we can use a unbound_range to avoid checking in order to unify the check logic

Reply:
> can we use the filter chain instead of if else ?
> 
> just like key_range, we can use a unbound_range to avoid checking in order to unify the check logic

fixed